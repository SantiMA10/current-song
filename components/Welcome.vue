<template>
  <div class="h-full flex flex-col text-white">
    <div class="text-right p-5">
      <modal name="colorSelector" class="h-screen">
        <chrome-picker v-model="color" class="z-50" @input="updateColor" />
      </modal>
      <button @click="show()">Color</button>
    </div>
    <div class="h-full flex items-center justify-center flex-col">
      <h1 class="text-3xl">
        üéµ Current song üéµ
      </h1>
      <p class="ml-8 mr-8 pt-4 pb-4">
        Login with Spotify and get an URL you can add to OBS (or other streaming
        programs) to show your audiciente your current Spotify song.
      </p>
      <LoginWithSpotify />
      <footer class="text-white pt-4">
        Made with ‚ù§ by <a href="https://santiagomartin.dev">@SantiMA10</a>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import VModal from 'vue-js-modal';
import { Chrome } from 'vue-color';

import LoginWithSpotify from '~/components/LoginWithSpotify.vue';

Vue.use(VModal, { dialog: true });

export default Vue.extend({
  components: {
    LoginWithSpotify,
    'chrome-picker': Chrome,
  },
  props: {
    customColor: {
      required: true,
      type: String,
    },
  },
  data() {
    return {
      color: this.$props.customColor,
    };
  },
  methods: {
    show() {
      this.$modal.show('colorSelector');
    },
    hide() {
      this.$modal.hide('colorSelector');
    },
    updateColor({ hex }) {
      this.$emit('updateColorValue', hex);
    },
  },
});
</script>

<style>
.v--modal {
  background-color: transparent;
  display: flex;
  flex-direction: column;
  justify-content: center;
  place-items: center;
  box-shadow: none;
}
</style>
