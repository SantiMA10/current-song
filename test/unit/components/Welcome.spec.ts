import { mount } from '@vue/test-utils';
import Welcome from '~/components/Welcome.vue';

describe('Welcome', () => {
  it('renders', () => {
    const subject = mount(Welcome);

    expect(subject.element).toMatchSnapshot();
  });
});
