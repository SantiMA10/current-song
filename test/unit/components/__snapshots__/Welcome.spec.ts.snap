// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders 1`] = `
<div
  class="text-white flex items-center justify-center flex-col"
>
  <h1
    class="text-3xl"
  >
    
    üéµ Current song üéµ
  
  </h1>
   
  <p
    class="ml-8 mr-8 pt-4 pb-4"
  >
    
    Login with Spotify and get an URL you can add to OBS (or other streaming
    programs) to show your audiciente your current Spotify song.
  
  </p>
   
  <a
    class="bg-green-500 p-2 rounded-full m-2 hover:bg-green-400 shadow"
    href="https://accounts.spotify.com/authorize?response_type=token&client_id=undefined&scope=user-read-currently-playing&redirect_uri=http%3A%2F%2Flocalhost"
  >
    
  Login with Spotify

  </a>
   
  <footer
    class="text-white pt-4"
  >
    
    Made with ‚ù§ by 
    <a
      href="https://santiagomartin.dev"
    >
      @SantiMA10
    </a>
  </footer>
</div>
`;
